libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI+Blocks.m

```diff
@@ -231,23 +231,27 @@ - (SFRestRequest *) performRequestWithMethod:(SFRestMethod)method path:(NSString
 #pragma mark - response delegate
 
 - (void) sendActionForRequest:(SFRestRequest *)request success:(BOOL)success withObject:(id)object {
-    if( success ) {
-        // This block def basically generalizes the SFRestDictionaryResponseBlock and SFRestArrayResponseBlock
-        // block typedefs, so that we can handle either.
+
+    // Ensures that the response is a well-known type.
+    if (![object isKindOfClass:[NSDictionary class]] && ![object isKindOfClass:[NSArray class]] && ![object isKindOfClass:[NSString class]]) {
```

Comment:
I got to the bottom of the issue. The response here is an empty `NSData` object, the debugger shows it as `<>`, but it's really an empty instance of `NSData`. It fell through the cracks because we have a `nil` check but no check for an empty object. I should have a better fix soon.